### PR TITLE
Added line graphs and pie charts for event tracking

### PR DESCRIPTION
**Pull request by molliemac**
I integrated your guard clause suggestion in events_controller.rb, and began my spec for that controller - still needs work. The line graphs are appearing, but I'm not sure if I correctly installed Postgres:/

### CODE REVIEW

File: app/controllers/api/events_controller.rb [hunk 1]
```diff
@@ -12,16 +12,14 @@ def set_access_control_headers
    def create
    	registered_application = RegisteredApplication.find_by(url: request.env['HTTP_ORIGIN'])
 
-   	if registered_application.nil?
-   		render json: { error: "Unregistered application"}, status: :unprocessable_entity
-   	else
-   		@event = registered_application.events.build(event_params)
-   		if @event.save
+   	# Incorporated Guard Clause
+   	return render json: { error: "Unregistered application"}, status: :unprocessable_entity if registered_application.nil?
```
Comment (durhamka) [inline app/controllers/api/events_controller.rb:16]: Nice job!

File: app/views/registered_applications/show.html.erb [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
 <h1><%= @app.name %></h1>
-<h2><%= @app.url %></h2>
+<h2>URL: <%= @app.url %></h2>
 <h3> Events </h3>
-  <% @events.each do |event, count| %>
-  <h4><%= "#{event}" + ": " + "#{@app.events.count}" %><%=  %></h4>
-<% end %>
+  <%= pie_chart @app.events.group(:name).count %>
+  <%= line_chart @app.events.group_by_week(:created_at, last: 4).count %>
```
Comment (durhamka) [inline app/views/registered_applications/show.html.erb:7]: Nice job!

File: spec/controllers/events_controller_spec.rb [hunk 3]
```diff
@@ -2,4 +2,15 @@
 
 RSpec.describe EventsController, type: :controller do
 
+	# POST create. Make sure a new event gets created with a name when JS snippet is triggered
+	describe "POST create" do
+		it "increases the number of events by 1" do
```
Comment (durhamka) [inline spec/controllers/events_controller_spec.rb:7]: I would only test that this returns a 200!

**Review by durhamka — APPROVED** [review]
Looks good! Minor comments.

**Comment by molliemac** [comment]
Thanks Kinsey! I updated my events controller spec and committed the change
to this same user story - do I push to that branch as well or does the
commit alone show my changes?

**Pull request merged**